runtime/tools/python/ttrt/utils/check_ttrt_output.py

```diff
@@ -0,0 +1,126 @@
+            tile = self.tilized[self.__iter_tile_index]
+            self.__iter_tile_index += 1
+            return tile
+        else:
+            raise StopIteration("No more tiles available")
+
+    def __tilize(self) -> list:
+        tiles = []
+
+        r, c = self.shape[0], self.shape[1]
+        tile_height, tile_width = Tile.shape()
+
+        # Walk a window [tile_height, tile_width] over tensor/np.array and collect
+        # tiles.
+        for i in range(0, r, tile_height):
+            for j in range(0, c, tile_width):
+                t = self.data[i : i + tile_height, j : j + tile_width]
+                tile = Tile(t, (i // tile_height, j // tile_width))
+                tiles.append(tile)
+
+        return tiles
+
+
+class Comparator:
+    def __init__(
+        self, golden_function, comparison_function, comparison_function_additional_args=None
+    ):
+        self.golden_function = golden_function
+        self.comparison_function = comparison_function
+        self.comparison_function_additional_args = comparison_function_additional_args
+
+    def compare(self):
+        print("Running comparison...\n")
+        return self.__compare_tile_by_tile()
+
+    def __compare_tile_by_tile(self):
+        assert len(device_output_tensors) == 1, "Expected only one output tensor!"
+
+        inputs = [Tensor(input_tensor) for input_tensor in device_input_tensors]
+        output = Tensor(device_output_tensors[0])
+        # Execute golden on underlying np.arrays and wrap returned np.array into a Tensor.
+        golden = Tensor(self.golden_function(*device_input_tensors))
+
+        for input in inputs:
+            assert (
+                input.num_tiles == output.num_tiles == golden.num_tiles
+            ), "Expected same number of tiles in input, output and golden tensors for eltwise functions!"
+
+        comparison_passed_for_all_tiles = True
+        mismatched_tiles = []
+
+        for _ in range(output.num_tiles):
+            output_tile = output.get_next_tile()
+            golden_tile = golden.get_next_tile()
+
+            if comparison_function_additional_args:
+                comparison = self.comparison_function(
+                    output_tile.data, golden_tile.data, **self.comparison_function_additional_args
+                )
+            else:
+                comparison = self.comparison_function(output_tile.data, golden_tile.data)
+
+            if comparison == False:
+                comparison_passed_for_all_tiles = False
+                mismatched_tiles.append(output_tile.index)
+
+            print(f"Tile:\n {output_tile.index}")
+            print(f"Device output:\n {output_tile}")
+            print(f"Golden output:\n {golden_tile}")
+            print(f"Comparison:\n {comparison}\n")
+
+        return comparison_passed_for_all_tiles, mismatched_tiles
+
+
+if __name__ == "__main__":
+    # ---- Set these to your desire ----
+    golden_function = np.add
```

Comment:
Maybe we could put these functions as arguments we can pass to a script. Not really pushing to do this now, but maybe just add a TODO